### DISCUSSION TITLE
Broadcast skin change to all players

### DISCUSSION THREAD

**Discussion by TheLeCrafter** [Minestom/Minestom#1370] [Help]
Hello, I am setting the player skin and currently it only shows the new skin for the player and Steve or Alex for the other players. I now want to broadcast this skin change to all players. What would be the best method for that?

**Comment by GreatWyrm** [comment]
What are you currently using to set the Player Skin?

I would recommend [Player#setSkin](https://javadoc.minestom.net/net/minestom/server/entity/Player.html#setSkin(net.minestom.server.entity.PlayerSkin)) if you are not currently using that.

**Reply by TheLeCrafter** [reply]
I am already using that

**Reply by GreatWyrm** [reply]
setSkin already broadcasts the skin change to other players, though. You may be using the method incorrectly, would you mind posting what code you have/create a video demonstrating this?

**Reply by TheLeCrafter** [reply]
It's Kotlin, so my code is:
```kotlin
player.skin = PlayerSkin.fromUuid(PlayerSkinConstants.DEFAULT_SKIN_UUID)
```

**Reply by GreatWyrm** [reply]
I'm afraid I don't know Kotlin that well, so I'd recommend asking someone who does
But from what I can tell, you're setting the skin data variable directly. The setSkin function does more than simply setting the variable, it also creates and broadcasts a number of packets to refresh the skin for other players.

**Reply by DigitalJokerMan** [reply]
Kotlin will map `player.skin = ...` to `player.setSkin(...)`. The method should be getting called.
